docs/CodeOverview.md

```diff
@@ -0,0 +1,21 @@
+# Code OverView
+
+## EntryPoint and Folder Structure
+
+TODO: add description of entry point and repo structure
+
+## Code Style
+
+TODO: are we using some code style?
+
+## Class Diagram
+
+TODO: relationship of some important classes
```

Comment:
Yes I found the title  "Class Diagram" is not good. Diagram is just a tool to describe something. The original idea is to put some brief overall class relationship here. But I think as you said it can be merged to each context such as "Architectural Overview".

I am using IntelliJ as IDE so when want to see class relationship of specified one I just use https://www.jetbrains.com/help/idea/viewing-class-hierarchy-as-a-class-diagram.html
But when need to create some more abstract or brief diagram I just use draw.io